ocaml/idl/json_backend/gen_json.ml

```diff
@@ -381,6 +381,32 @@ end = struct
     else
       cmp
 
+  let rec list_dedup cmp =
```

Comment:
I think we have an implementation of this elsewhere already (somewhere in stdext)?
Although there is `List.sort_uniq` would that help, or you need more control over which one to keep?